src/fb/nack.rs

```diff
@@ -0,0 +1,272 @@
+                    let ret = base.wrapping_add(self.mask_i as u16 - 1);
+                    return Some(ret);
+                }
+                self.mask_i += 1;
+                if self.mask_i > 15 {
+                    break;
+                }
+            }
+        }
+    }
+}
+
+/// Generic NACK FCI information as specified in RFC 4585
+pub struct Nack<'a> {
+    data: &'a [u8],
+}
+
+impl<'a> Nack<'a> {
+    /// The list of RTP sequence numbers that is being NACKed.
+    pub fn entries(&self) -> impl Iterator<Item = u16> + '_ {
+        NackParserEntryIter {
+            parser: self,
+            i: 0,
+            mask_i: 0,
+        }
+    }
+
+    pub fn builder() -> NackBuilder {
+        NackBuilder { rtp_seq: vec![] }
+    }
+}
+
+impl<'a> FciParser<'a> for Nack<'a> {
+    const PACKET_TYPE: FciFeedbackPacketType = FciFeedbackPacketType::TRANSPORT;
+    const FCI_FORMAT: u8 = 1;
+
+    fn parse(data: &'a [u8]) -> Result<Self, RtcpParseError> {
+        Ok(Self { data })
+    }
+}
+
+pub struct NackBuilderEntryIter<'a> {
+    builder: &'a NackBuilder,
+    i: usize,
+    base_entry: u16,
+}
+
+impl<'a> NackBuilderEntryIter<'a> {
+    fn encode_entry(base: u16, mask: u16) -> [u8; 4] {
+        [
+            ((base & 0xff00) >> 8) as u8,
+            (base & 0xff) as u8,
+            ((mask & 0xff00) >> 8) as u8,
+            (mask & 0xff) as u8,
+        ]
+    }
+}
+
+impl<'a> std::iter::Iterator for NackBuilderEntryIter<'a> {
+    type Item = [u8; 4];
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut bitmask = 0;
+        loop {
+            if self.i >= self.builder.rtp_seq.len() {
+                if self.i > 0 {
+                    let last_entry = self.builder.rtp_seq[self.i - 1];
+                    if self.base_entry != last_entry {
+                        // we need to output the final entry
+                        let ret = NackBuilderEntryIter::encode_entry(self.base_entry, bitmask);
+                        self.base_entry = last_entry;
+                        return Some(ret);
+                    }
+                }
+                return None;
+            }
+
+            let entry = self.builder.rtp_seq[self.i];
+            if self.i == 0 {
+                // initial set up for the first value
+                self.base_entry = entry;
+                self.i += 1;
+                continue;
+            }
+            let diff = entry.wrapping_sub(self.base_entry);
+            if diff > 0 {
+                bitmask |= 1 << (diff - 1);
+            }
+            if diff >= 16 {
+                // bitmask will overflow next iteration, return the current value
+                let ret = NackBuilderEntryIter::encode_entry(self.base_entry, bitmask);
+                self.base_entry = entry;
+                return Some(ret);
+            }
+            self.i += 1;
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NackBuilder {
+    rtp_seq: Vec<u16>,
```

Comment:
Suggestion: use a [`BTreeSet`](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html) so we don't need to `sort()` when adding an RTP ts.